src/ui/tournament/detail/joinForm.tsx

```diff
@@ -34,19 +35,29 @@ function close(fromBB?: string) {
 }
 
 function renderForm() {
+  const t = tournamentCtrl.tournament
+  const teamList = t.teamBattle ? t.teamBattle.joinWith.map((x: string) => [t.teamBattle ? t.teamBattle.teams[x] : '', x]) : []
+
+  if (t.teamBattle && !t.teamBattle.joinWith.includes(settings.tournament.join.lastTeam())) {
```

Comment:
I find it weird to have this logic in a view. It should be in the controller instead.